Login bug

HalahRaadSalih opened the pull request:
Implemented a fix for db production.sql file that was sending empty line for quotes created_at updated_at time stamps. Also, modified index.html after the user login to check if the quote has a created_at value before displaying the associated div. Modified test to check for created_at values and their conversions.

HalahRaadSalih committed:
fixed login bug

HalahRaadSalih committed:
added test cases for dates

HalahRaadSalih committed:
changed migration to prevent null dates

tebba-von-mathenstein commented on .vagrant/machines/default/virtualbox/action_provision:
```diff
@@ -0,0 +1 @@
+1.5:89e0e1f7-be8b-4fc5-a492-db089f53516e
```
Lets add everything in .vagrant to the .gitignore please. These don't belong in version control

tebba-von-mathenstein commented on Gemfile.lock:
```diff
@@ -47,27 +47,27 @@ GEM
     execjs (2.2.2)
     hike (1.2.3)
-    i18n (0.6.11)
+    i18n (0.7.0)
```
Why do we need to update i18n? This is for localization and we're locked on 0.6.11 for `reasons`

tebba-von-mathenstein commented on Gemfile.lock:
```diff
@@ -47,27 +47,27 @@ GEM
-    minitest (5.4.2)
-    multi_json (1.10.1)
+    minitest (5.8.4)
+    multi_json (1.11.2)
```
These other updates, why do we need them ?

tebba-von-mathenstein commented on Vagrantfile:
```diff
@@ -0,0 +1,71 @@
+# -*- mode: ruby -*-
```
This should also be in .gitignore please

tebba-von-mathenstein commented on app/helpers/application_helper.rb:
```diff
@@ -1,2 +1,3 @@
 module ApplicationHelper
+  
```
Please revert this unneeded space.

tebba-von-mathenstein commented on app/views/quotes/index.html.erb:
```diff
@@ -8,8 +8,11 @@
+      <div class="created">
+          <%= time_ago_in_words(quote.created_at) %> ago
+      </div>
+    <% end %>
```
I think we should add a case for if `quote.created_at` is `nil` -- lets just display todays date instead!

tebba-von-mathenstein commented on db/migrate/20140712034016_add_timestamps_to_quotes.rb:
```diff
@@ -1,7 +1,7 @@
   def change
     change_table :quotes do |t|
-      t.timestamps
+      t.timestamps null: false
```
Hey, this is the right idea -- but this migration has already run in production so to make the database accept the new constraint we actually need to create a new migration, otherwise rails will just assume this one has run, and not change the database.

tebba-von-mathenstein commented on db/production.sql:
```diff
@@ -8,8 +8,8 @@ SET client_min_messages = warning;
-1	People who think they know everything are a great annoyance to those of us who do.	Isaac Asimov	\N	\N
-2	As a child my family's menu consisted of two choices: take it or leave it.	Buddy Hackett	\N	\N
+1	People who think they know everything are a great annoyance to those of us who do.	Isaac Asimov 2014-07-12 03:40:54.405448  2014-07-12 03:50:54.405448
+2	As a child my family's menu consisted of two choices: take it or leave it.	Buddy Hackett	2014-07-12 03:40:54.405448	2014-07-12 03:51:54.405448
```
Nice catch here on updating the data, thanks!

tebba-von-mathenstein commented on spec/features/quotes_spec.rb:
```diff
@@ -1,12 +1,16 @@
 require 'capybara/rails'
+require 'action_view'
+require 'action_view/helpers'
+require 'date'
```
Hey, I don;t see this being used below? Am I mistaken?

tebba-von-mathenstein replied:
Halah says: "Date is for Date.today used below, and action_view/helpers was in a test that I decided to remove. I'll remove those :) "

tebba-von-mathenstein replied:
Awesome, thanks.

tebba-von-mathenstein commented:
Hey nice work here -- before I merge this code please remove all the vagrant files from the codebase and put them in your global gitignore. Also see my inline comments.

HalahRaadSalih committed:
added vagrant files to gitignore

HalahRaadSalih committed:
added Vagrantfile

HalahRaadSalih committed:
added migration to modidy the quotes timestamp coloum to not to accept null values

tebba-von-mathenstein commented:
Hey I fixed the .gitignore on my end since this is preventing our users from logging on in production right now. I'm going to go ahead and merge this code, thanks for getting this done.

tebba-von-mathenstein commented:
Reopened at your request

HalahRaadSalih committed:
removed unnecessary require statements from test file

HalahRaadSalih commented:
When a user registers, they should be logged in automatically, but they were lead to the login page again, so I've fixed the user_id field name in the registration controller. Also, I still have problem the json gem, so I've used 1.8.3 instead of 1.8.1.

HalahRaadSalih commented:
I've added vagarant files to gitignore, created new migration to add restriction to the quotes time stamp coloum. Also, I've removed unnecessary require statements.

tebba-von-mathenstein commented:
I think you may need to use 
`git rm --cached -r .vagrant` to remove the vagrant files from the commit history.

HalahRaadSalih committed:
removed vagrant files

HalahRaadSalih commented:
All set. :)

tebba-von-mathenstein commented:
Wonderbar, thanks again!
